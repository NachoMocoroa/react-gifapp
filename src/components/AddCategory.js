import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const placeholderText = 'Search and Enter';
    const [inputValue, setInputValue] = useState('');
    const handleChange = (e) => {
        setInputValue(e.target.value);
    };
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 1) {
            setCategories([inputValue]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={ handleSubmit }>
            <div className="input-box">
                <input
                    type="text" 
                    className="underlined" 
                    placeholder={ placeholderText } 
                    value={ inputValue } 
                    onChange={ handleChange } 
                />
                <span className="highlight"></span>
            </div>
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};

export default AddCategory;
