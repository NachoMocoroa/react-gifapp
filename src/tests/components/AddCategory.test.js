import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Test AddCategory', () => {
    
    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('should render AddCategory component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should not call "setCategories" on submit empty value', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call "setCategories" on submit and clear input', () => {
        const value = 'gif';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Array));
        expect(wrapper.find('input').prop('value')).toBe('');
    });

});
