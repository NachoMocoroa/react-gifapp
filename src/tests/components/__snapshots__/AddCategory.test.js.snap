// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddCategory should render AddCategory component 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-box"
  >
    <input
      className="underlined"
      onChange={[Function]}
      placeholder="Search and Enter"
      type="text"
      value=""
    />
    <span
      className="highlight"
    />
  </div>
</form>
`;
